pkg/consensus/consensus.go

```diff
@@ -239,9 +241,16 @@ func (s *service) newPrepareRequest() payload.PrepareRequest {
 			panic(err)
 		}
 	}
+	r.nonce = s.getNonce()
 	return r
 }
 
+func (s *service) getNonce() uint64 {
+	b := make([]byte, 8)
+	_, _ = rand.Read(b)
```

Comment:
It technically can return an error.

Reply:
Yes, and ignoring this (highly unlikely) error seems like a good way to deal with it. May be we could fallback to the `math/rand` though.

Reply:
panic!

Reply:
Added fallback.

Reply:
Using `math/rand` here is just not appropriate. If the node can't produce proper random number it better just die and let others do the job.